src/mappings/contracts.ts

```diff
@@ -390,33 +411,26 @@ export async function contractUpdateUsedResources(
 ) {
   const usedResources = new SmartContractModuleUpdatedUsedResourcesEvent(ctx, item.event).asV49
 
-  const contractUsedResources = new ContractResources()
-
-  const savedContract = await ctx.store.get(NodeContract, { where: { contractID: usedResources.contractId } })
+  const savedContract = await ctx.store.get(Contract, { where: { contractID: usedResources.contractId } })
   if (!savedContract) return
 
-  const savedContractResources = await ctx.store.get(ContractResources, { where: { contract: { contractID: savedContract.contractID } }, relations: { contract: true } })
-  if (savedContractResources) {
-    savedContractResources.cru = usedResources.used.cru
-    savedContractResources.sru = usedResources.used.sru
-    savedContractResources.hru = usedResources.used.hru
-    savedContractResources.mru = usedResources.used.mru
-    await ctx.store.save<ContractResources>(savedContractResources)
-
-    savedContract.resourcesUsed = savedContractResources
-    await ctx.store.save<NodeContract>(savedContract)
-  } else {
-    contractUsedResources.id = item.event.id
-    contractUsedResources.cru = usedResources.used.cru
-    contractUsedResources.sru = usedResources.used.sru
-    contractUsedResources.hru = usedResources.used.hru
-    contractUsedResources.mru = usedResources.used.mru
-    contractUsedResources.contract = savedContract
-    await ctx.store.save<ContractResources>(contractUsedResources)
-
-    savedContract.resourcesUsed = contractUsedResources
-    await ctx.store.save<NodeContract>(savedContract)
+  if (savedContract.nodeID) {
+    const savedNode = await ctx.store.get(Node, { where: { nodeID: savedContract.nodeID } })
+    if (!savedNode) return
+
+    savedNode.freeMRU += (savedContract.usedMRU || BigInt(0)) - usedResources.used.mru
+    savedNode.freeHRU += (savedContract.usedHRU || BigInt(0)) - usedResources.used.hru
+    savedNode.freeSRU += (savedContract.usedSRU || BigInt(0)) - usedResources.used.sru
+
+    await ctx.store.save<Node>(savedNode)
```

Comment:
As previous comment, should be safer to use `.increment()` or `.decrement()`. check [here](https://github.com/typeorm/typeorm/blob/master/docs/repository-api.md#:~:text=increment%20%2D%20Increments%20some%20column%20by%20provided%20value%20of%20entities%20that%20match%20given%20options.)